commands/generate.go

```diff
@@ -1,23 +1,118 @@
 package commands
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
 	"github.com/spf13/cobra"
 )
 
 func GenerateCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "generate",
+		Use:   "generate <file name>",
 		Short: "Creates new migrations",
+		Long:  "morph generate create_users --driver postgresql --dir db/migrations --timestamp",
 		Args:  cobra.ExactArgs(1),
 		Run:   generateCmdF,
 	}
 
-	cmd.Flags().StringP("driver", "d", "", "the driver to use")
-	cmd.MarkFlagRequired("driver")
+	cmd.Flags().StringP("driver", "d", "", "the driver to use.")
+	cmd.Flags().BoolP("timestamp", "t", false, "a timestamp prefix will be added to migration file if set.")
+	cmd.Flags().StringP("timeformat", "f", "unix", "timestamp format to be used for timestamps.")
+	cmd.Flags().StringP("timezone", "z", "utc", "time zone to be used for timestamps.")
+	cmd.Flags().BoolP("sequence", "s", false, "a sequence number prefix will be added to migration file if set.")
+	cmd.Flags().IntP("number-of-digits", "n", 6, "number of digits to be used in sequence prefix.")
+	_ = cmd.MarkFlagRequired("driver")
 
 	return cmd
 }
 
-func generateCmdF(cmd *cobra.Command, _ []string) {
-	cmd.Println("To Be Implemented")
+func generateCmdF(cmd *cobra.Command, args []string) {
+	dir, _ := cmd.Flags().GetString("dir")
+	extention := "sql"
+	fileName := args[0]
+
+	if ts, _ := cmd.Flags().GetBool("timestamp"); ts {
+		tz, _ := cmd.Flags().GetString("timezone")
+		loc, err := time.LoadLocation(tz)
+		if err != nil {
+			cmd.PrintErrln(err)
+			return
+		}
+		date := time.Now().In(loc)
+
+		tf, _ := cmd.Flags().GetString("timeformat")
+		switch tf {
+		case "unix":
+			fileName = strings.Join([]string{strconv.FormatInt(date.Unix(), 10), fileName}, "_")
+		case "unix-nano":
+			fileName = strings.Join([]string{strconv.FormatInt(date.UnixNano(), 10), fileName}, "_")
+		default:
+			fileName = strings.Join([]string{date.Format(tf), fileName}, "_")
+		}
+	} else if seq, _ := cmd.Flags().GetBool("sequence"); seq {
+		numDigits, _ := cmd.Flags().GetInt("number-of-digits")
```

Comment:
I think we can move that responsibility on the tool itself, and do not allow people to configure the digit number, in a sense we are presenting a more opinionated approach to seq generation. 3/5 on this, just feels to me like exposing an internal concern for configuration. Thoughts?

Reply:
Makes sense, I don't have strong opinions 😅

Reply:
Same, and going with a default of 6, you have to add **a lot** of migrations before this becoming a concern, so I think it's safe